### PR TITLE
Cleanup dependencies

### LINKED ISSUE #153: Drop dependency on App Engine SDK
There is no good reason to have the App Engine SDK as the implementation dependency in our libraries.

The SDK itself provides APIs that one can use *only* in the App Engine 8 environment and has an enormous 15mb size. Also, while in many projects we're going forward with Java 11 and App Engine 11, we'll no longer be able to use the App Engine SDK APIs.

### PR DESCRIPTION
**Pull request by yuri-sergiichuk**
The goal of this PR is to resolve #153 as well as upgrade our dependencies for the UI libraries. The removed AppEngine SDK was not used in any of the modules.

As part of the PR, I have addressed various deprecations in the NodeJS environment and related required changes in JS tests.

### CODE REVIEW

**Comment by yuri-sergiichuk** [comment]
@dmdashenkov @dmitrykuzmin PTAL.

File: client-js/main/client/actor-request-factory.js [hunk 1]
```diff
@@ -815,7 +815,7 @@ class QueryFactory {
    */
   static _newId() {
     const result = new QueryId();
-    result.setValue(`q-${uuid.v4()}`);
+    result.setValue(`q-${uuidv4()}`);
```
Comment (dmdashenkov) [inline client-js/main/client/actor-request-factory.js:818]: Is this some kind of a convention? Why do we alias this particular import?

**Review by dmdashenkov — CHANGES_REQUESTED** [review]
@yuri-sergiichuk, please see my comment. Also, was there no use of AppEngine SDK at all? If so, please mention this in the PR description.

File: client-js/main/client/actor-request-factory.js (hunk 1 above)
Comment (yuri-sergiichuk) [reply client-js/main/client/actor-request-factory.js:818]: The default exported item is `v4` and I don't think it worth using it as is. That's why I used the alias.

With just `q-${v4()}` it is more awkward.

File: integration-tests/js-tests/test/firebase-client/given/firebase-client.js [hunk 2]
```diff
@@ -37,7 +35,7 @@ import TestEnvironment from "../../given/test-environment";
  * @return {FirebaseClient} the Firebase client instance
  */
 export function initClient(endpointUrl, tenantProvider, keepUpInterval) {
-  return spineWeb.init({
+  return initSpineWeb({
```
Comment (dmitrykuzmin) [inline integration-tests/js-tests/test/firebase-client/given/firebase-client.js:38]: Maybe `initSpineWebClient`?

File: integration-tests/js-tests/test/firebase-client/given/firebase-client.js [hunk 3]
```diff
@@ -37,7 +35,7 @@ import TestEnvironment from "../../given/test-environment";
  * @return {FirebaseClient} the Firebase client instance
```
[the remark this answers is not in the document]
Comment (yuri-sergiichuk) [reply integration-tests/js-tests/test/firebase-client/given/firebase-client.js:35]: I personally totally am. Mine shows warnings for unresolved types when the type is not imported. Also, JS through webpack and Babel has tree-shaking that'll eliminate all parts that are not being used in the runtime.

**Review by dmitrykuzmin — APPROVED** [review]

**Review by dmdashenkov — APPROVED** [review]

**Pull request merged**